### PR TITLE
feat(audit): set lastEditorId when changing auditable entities

### PR DESCRIPTION
**Pull request by Phil91**
## Description

lastEditorId is now set whenever a auditable entity is changed 
index for lastEditorId columns created

## Why

to be able to see who changed a auditable entity

## Issue

N/A - Jira Ticket: CPLP-2960

## Corresponding CD PR

[#54](https://github.com/eclipse-tractusx/portal-cd/pull/54)

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Phil91** [comment]
@ntruchsess the latest changes are in

File: src/administration/Administration.Service/Controllers/CompanyDataController.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ public async Task<NoContentResult> CreateCompanyRoleAndConsentAgreementDetailsAs
     /// <remarks>Example: Get: api/administration/companydata/certificateTypes</remarks>
     /// <response code="200">Returns a collection of certificates.</response>
     [HttpGet]
-    [Authorize(Roles = "request_ssicredential")]
+    // [Authorize(Roles = "request_ssicredential")]
```
Comment (ntruchsess) [inline src/administration/Administration.Service/Controllers/CompanyDataController.cs:196]: have those authorizations been removed on purpose?

File: src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs [hunk 2]
```diff
@@ -36,13 +37,18 @@ namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities;
 /// </remarks>
 public class PortalDbContext : DbContext
 {
+    private readonly IIdentityService _identityService;
+
     protected PortalDbContext()
     {
+        Console.WriteLine("Constructor without params hit");
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs:44]: maybe throwing InvalidOperationException is more suitable? The console-output might not be visible anywhere.

**Review by ntruchsess — CHANGES_REQUESTED** [review]
I did some cleanup removing the remaining writes to LastEditorId and some redundant uses of WithUserIdAndCompanyId that have been added lately. Please double-check. In case there were good reasons for those calls that I just don't see let's discuss this.

File: src/administration/Administration.Service/Controllers/CompanyDataController.cs (hunk 1 above)
Comment (Phil91) [reply src/administration/Administration.Service/Controllers/CompanyDataController.cs:196]: that wasn't on purpose, i did rollback the removal

File: src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs (hunk 2 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/PortalDbContext.cs]: Agreed, i did change the implementation

**Comment by Phil91** [comment]
@ntruchsess all changes are in, please be aware of the cd pr [#54](https://github.com/eclipse-tractusx/portal-cd/pull/54)

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**